File: Java/lista_2/src/Main.java
public class Main {
    public static void main(String[] args) {
        try {
            if (args.length == 0) {
//                throw new ArabRzymException("Napisz liczbę arabską lub rzymską jako argument.");
                for (int i = 1; i <= 4000; i++) {
                    String rzymNumber = ArabRzym.arab2rzym(i);
                    System.out.println(i + " -> " + rzymNumber);
                    System.out.println(rzymNumber + " -> " + ArabRzym.rzym2arab(rzymNumber));
                }
                return;
            }

            String input = args[0].toUpperCase();

            if (ArabRzym.isValidRoman(input)) {
                int result = ArabRzym.rzym2arab(input);
                System.out.println(input + " -> " + result);
            } else if (ArabRzym.isValidArabic(input)) {
                int arabic = Integer.parseInt(input);
                String result = ArabRzym.arab2rzym(arabic);
                System.out.println(input + " -> " + result);
            } else {
                throw new ArabRzymException("Nieprawidłowa dana: " + input);
            }

        } catch (ArabRzymException e) {
            System.out.println("Error: " + e.getMessage());
//        } catch (NumberFormatException e) {
//            System.out.println("Błąd: Nieprawidłowy format liczby arabskiej.");
        }
    }
}

File: Cpp/lista_2/main.cpp
#include <iostream>
#include "ArabRzym.h"

int main(int argc, char *argv[]) {
    if (argc < 2) {
        throw ArabRzymException("Napisz liczbę arabską lub rzymską jako argument.");
    }
    for (int i = 1; i < argc; ++i) {
        try {
            std::string input = argv[i];
            for (char &c: input) c = toupper(c);

            if (ArabRzym::isValidRzym(input)) {
                int result = ArabRzym::rzym2arab(input);
                printf("%s -> %i\n", input.c_str(), result);
            } else if (ArabRzym::isValidArab(input)) {
                int arab = std::stoi(input);
                std::string result = ArabRzym::arab2rzym(arab);
                printf("%i -> %s\n", arab, result.c_str());
            } else {
                throw ArabRzymException("Nieprawidłowa dana: " + input);
            }
        } catch (ArabRzymException e) {
            std::cout << "Error: " << e.message << std::endl;
        }
    }

    return 0;
}
